### PR TITLE
Update Vault, end reliance on removed item API

### PR DESCRIPTION
**Pull request by PikaMug**
https://github.com/MilkBowl/VaultAPI/releases/tag/1.7

Vault 1.7 has removed the ItemAPI which had fallen apart with the release of MC 1.13. Given this version of Vault has now been released on both BukkitDev and SpigotMC, it's what most people will be attempting to use in conjunction with Magic.

The alternative to this would be replacing with something like the [XMaterial library](https://www.spigotmc.org/threads/1-8-to-1-13-itemstack-material-version-support.329630/).

### CODE REVIEW

File: Magic/pom.xml [hunk 1]
```diff
@@ -804,5 +804,40 @@
                 </executions>
             </plugin>
         </plugins>
+        <pluginManagement>
```
Comment (killme) [inline Magic/pom.xml]: Please make a separate commit / pr for this

File: Magic/src/main/java/com/elmakers/mine/bukkit/integration/VaultController.java [hunk 2]
```diff
@@ -102,10 +97,4 @@ public boolean depositPlayer(Player player, double amount) {
...
         return response.transactionSuccess();
     }
-
-    @Nullable
-    public String getItemName(Material material, short data) {
-        ItemInfo info = Items.itemByType(material, data);
-        return info == null ? null : info.getName();
-    }
-}
+}
```
Comment (killme) [inline Magic/src/main/java/com/elmakers/mine/bukkit/integration/VaultController.java]: Missing newline at end of file?
(`mvn checkstyle` should warn for this I think)

File: Magic/pom.xml [hunk 3]
```diff
@@ -85,7 +85,7 @@
         <dependency>
             <groupId>net.milkbowl.vault</groupId>
             <artifactId>VaultAPI</artifactId>
-            <version>1.6</version>
+            <version>1.7</version>
```
Comment (killme) [inline Magic/pom.xml]: No major version bump? Seriously Vault?

**Comment by NathanWolf** [comment]
I am wondering if we should just roll this and the new EssentialsX API stuff up into a version 8 of Magic, to eventually become 13+ only.

I am hesitant to remove features or outright break older versions, I bet lots of people still using the "old" Vault and Essentials.

File: Magic/pom.xml (hunk 1 above)
Comment (PikaMug) [reply Magic/pom.xml]: Whoops, didn't think that made it into the PR. Reason for that is Eclipse complains about "run" not being a valid goal for the ant plugin if the pluginManagement tag isn't present.

Anyway, I'll remove it.

File: Magic/src/main/java/com/elmakers/mine/bukkit/integration/VaultController.java (hunk 2 above)
Comment (PikaMug) [reply Magic/src/main/java/com/elmakers/mine/bukkit/integration/VaultController.java]: Didn't notice any warning but I'll restore it anyway!

File: Magic/pom.xml (hunk 3 above)
Comment (PikaMug) [reply Magic/pom.xml]: Sleaker = Slacker

Comment (NathanWolf) [reply Magic/pom.xml]: 😂

**Comment by PikaMug** [comment]
Any interest in this now that EssentialsX stuff is out?

**Comment by NathanWolf** [comment]
Yes, thanks! I decided not to make the leap to 8.0 but since we're now requiring latest EssentialsX I guess it's fine to ditch older Vault features, too.
